packages/docusaurus-theme-classic/src/theme/BackToTopButton/index.tsx

```diff
@@ -37,18 +37,29 @@ function smoothScrollTopNative(): CancelScrollTop {
 
 function smoothScrollTopPolyfill(): CancelScrollTop {
   let raf: number | null = null;
+  // Coercing to auto scroll to prevent conflicts with scroll-behavior.
+  // This is a polyfill for browsers without smooth scrolling anyways.
+  document.documentElement.style.scrollBehavior = 'auto';
```

Comment:
if a previous value is already set, will it be properly restored?

What's the impact of not restoring it?

Reply:
There should be no other case where a style is inlined in the DOM. I don't believe any userland code would do that instead of using `:root` selector. We need to force using `scroll-behavior: auto`, because otherwise CSS and JS would scroll the page simultaneously and cancel each other, as is the entire purpose of sending this PR.